lib/firenest/topology/erlang.ex

```
@@ -133,112 +98,327 @@ defmodule Firenest.Topology.Erlang do
     :ets.lookup_element(topology, :nodes, 2)
   end
 
-  ## GenServer callbacks
+  # Subscribing to the topology events is private right now,
+  # we can make it public if necessary but sync_named/3 should
+  # be enough for all purposes.
+  defp subscribe(topology, pid) when is_pid(pid) do
```

subscribe should return a monitor `ref` so that subscribers get `:DOWN` if the response isn't coming. There is 5 second timeout right now but it might be nice to fail faster for something that is straightforward.

Given the current use case its not an issue I think but if it was public there should be a way to uniquely identify the messages for a subscription. This could be done with subscription `ref` but again no need to complicate as private in process created just for this.